src/main/java/org/mars/rover/kata/entrydata/StdinProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.mars.rover.kata.entrydata;
+
+import org.mars.rover.kata.Direction;
+import org.mars.rover.kata.Position;
+import org.mars.rover.kata.commands.CommandParser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Scanner;
+
+public class StdinProcessor implements InputProcessorInterface {
+    CommandParser commandParser;
+
+    public StdinProcessor(CommandParser commandParser) {
+        this.commandParser = commandParser;
+    }
+
+
+    public CommandSet processInput(Iterator<String> stringIterator) {
```

Comment:
OK, tutaj widać, że to jest funkcja tzw. sedno sprawy. Robi tutaj kupę roboty. Parsuje kolejne linijki, nadaje im biznesowy sens i tworzy z nich biznesowe obiekty. Nie ma w tym nic inherentnie złego. Ba, nazwy zmiennych są czytelne, więc każdy początkujący programista może to prześledzić i zrozumieć co tu się dzieje.

To jest właśnie to programowanie algorytmiczne, o którym mówiłem. Tak uczą pisać kury programowania. Nie ma w tym nic złego. Po prostu cała uwaga i całe mięcho tej aplikacji dzieje się w tej jednej funkcji. Co oznacza, że będzie ona rosła i rosła, wraz z wprowadzeniem obsługi błędów czy dodawaniu nowych ficzerów. Będzie też piekielnie trudna do przetestowania, no bo robi całą robotę.

Na pierwszy rzut oka widać, żę jest za długa. Do tego jak mamy 4 poziomy wcięcia, do tego fora w którym jest zagniezdżone przetwarzanie strumienoiwe. A, i iteracja o +2, co zawsze wprowadza zamieszanie prędzej czy później.

Jest to myślę temat na refaktor na następnych zajęciach.

Jeszcze raz powtórzę — nie ma w tym nic niepoprawnego. Można to po prostu zapisać czytelniej, rozbić na mniejsze kawałki, które się łatwiej testuje, a co za tym idzie zmienia i naprawia.

Reply:
Dodane do release/1.0.